http.ts

```diff
@@ -206,7 +286,27 @@ async function readRequest(c: Conn): Promise<[ServerRequest, BufState]> {
 
   [req.headers, err] = await tp.readMIMEHeader();
 
-  // TODO: handle body
-
   return [req, err];
 }
+
+async function collect(
+  it: AsyncIterableIterator<Uint8Array>
```

Comment:
Hm. This is very similar to `deno.Buffer.readAll()` ...  That bothers me... 

(Also here there is a `buffer.ts` which duplicates functionality in the `deno.Buffer`. I will remove that.)

Reply:
I've removed buffer.ts now 5abec6efc5028bb281c29f5df40c0d87a4b8ebeb

Reply:
This might be a nice thing to have in `buffer.ts` of deno actually...

Reply:
Hm. I guess I'm wondering if `readChunk` couldn't instead be made a reader. I suppose the difference with `readChunk` is that the size isn't specified by the caller. Hm.
Well I need to think about this but it's fine for now.

Can we rename `collect` to something like `readAllIterator` or even `readAll2`? It's basically the same concept, so I think it ought to have a similar name.